src/Sets/Line.jl

```diff
@@ -323,11 +323,17 @@ function constrained_dimensions(L::Line{N}) where {N<:Real}
     return nonzero_indices(L.a)
 end
 
-function _linear_map_hrep(M::AbstractMatrix{N}, P::Line{N}, use_inv::Bool;
-                          inverse::Union{Nothing, AbstractMatrix{N}}=nothing
-                         ) where {N<:Real}
-    constraint = _linear_map_hrep_helper(M, P, use_inv; inverse=inverse)[1]
-    return Line(constraint.a, constraint.b)
+function _linear_map_hrep_helper(M::AbstractMatrix{N}, P::Line{N},
+                                 algo::AbstractLinearMapAlgorithm) where {N<:Real}
+    constraints = _linear_map_hrep(M, P, algo)
+    if length(constraints) == 1
+        c = first(constraints)
+        return Line(c.a, c.b)
+    else
+        # TODO: check and transform 2 constraints to a line? ie. add constructor
+        # from constraints? (see also the corresponding function for HalfSpace)
+        return HPolyhedron(constraints)
+    end
```

Comment:
I do not understand why this code is not the same as for `Hyperplane` since the `Line` is a `Hyperplane` in 2D. In particular, since the original constraints are identical up to the sign, it should not happen that one of them gets removed. There can probably be a common helper function to handle `Line` and `Hyperplane` together.

However, I agree that the old code was wrong in two respects:
1. It did not check whether the output dimension is 2.
2. It did not check whether the the map is a projection, which can destroy the property of being a hyperplane/line. Take the 2D example `x + y = 0` and project to the x axis, which gives you the 1D universe.

Reply:
ok, i have removed the TODO and kept 

```julia
function _linear_map_hrep_helper(M::AbstractMatrix{N}, P::Line{N},
                                 algo::AbstractLinearMapAlgorithm) where {N<:Real}
     constraints = _linear_map_hrep(M, P, algo)
     @assert length(constraints) == 2
     c = first(constraints)
     return Line(c.a, c.b)
end
```
in particular, what i meant is that we are not checking that the 2 constraints are of the form `ax <= b` and `ax >= b`, which implies `ax = b`.

Reply:
What about my second concern? I think that is really a problem:
> It did not check whether the map is a projection, which can destroy the property of being a hyperplane/line. Take the 2D example x + y = 0 and project to the x axis, which gives you the 1D universe.

Reply:
Not sure that i follow, do you mind opening a follow up issue? (no regression in this branch if you say it wasn't checked before..)

Reply:
#1989